www/profile/index.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+$servername = "localhost";
+$username = "worker";
+$password = "swiffcat";
+$db = "socialmedia";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $db);
+
+// Check connection
+if ($conn->connect_error) {
+    header("HTTP/1.1 500 Internal Server Error");
+    die();
+}
+
+$headers = getallheaders();
+if (!array_key_exists("token", $headers)) {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+$token = getallheaders()["token"];
+
+// If invalid header token, die
+$stmt = $conn->prepare("SELECT user_id FROM auth WHERE token = ?");
+$stmt->bind_param("s", $a1);
+$a1 = $token;
+$stmt->execute();
+$stmt->store_result();
+$accessor_id = null;
+$stmt->bind_result($accessor_id);
+if ($stmt->fetch()) {
+    // user_id is bound
+} else {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET') {
+    if (!isset($_GET["id"])) {
+        echo $_GET;
+        header("HTTP/1.1 403 Forbidden");
+        die();
+    }
+
+    $user_id = $_GET["id"];
+
+    // make sure the user exists and grab their data
+    $stmt = $conn->prepare("SELECT * FROM users WHERE user_id = ?");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $count = $result->num_rows;
+    if ($count != 1) {
+        header("HTTP/1.1 403 Forbidden");
+        die();
+    }
+    $row = $result->fetch_assoc();
+
+    // Retrieve data from the posts table
+    $stmt = $conn->prepare("SELECT * FROM posts WHERE user_id = ? ORDER BY created_at DESC");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $posts = array();
+
+    if ($result->num_rows > 0) {
+        while ($post_row = $result->fetch_assoc()) {
+            $posts[] = $post_row;
+        }
+    }
+
+    // add posts to the values obtained earlier
+    $row["posts"] = $posts;
+    $json = json_encode($row);
```

Comment:
Why did you make a variable for this just to print it and end

Reply:
what do you mean?
posts or row?

row contains the users profile data and then we retrieve the post data and put it with the user data to then echo out as a json.
If posts, I'm first making the array, filling it, and then giving it to row.

Reply:
oh you mean the json, yea I could just make it echo the function call result.